src/ansi-c/expr2c.cpp

```diff
@@ -3794,6 +3794,49 @@ std::string expr2ct::convert_with_precedence(
   else if(src.id()==ID_type)
     return convert(src.type());
 
+  else if(src.id() == ID_rol)
+  {
+    // AAAA rol n == (AAAA << n) | (AAAA >> (width(AAAA) - n))
```

Comment:
What if `n` is greater than bit width?

Reply:
Also note the cases when shift can give undefined behaviour.

Reply:
This is done with modulo now.